src/algorithms/bfgs.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <RcppArmadillo.h>
+#include "../families/family.h"
+#include "wolfe_line_search.h"
+
+using namespace Rcpp;
+using namespace arma;
+
+// // BFGS Algorithm.
+// // Solves argmin_z { f(z) + rho/2*||z-u||^2 }
+template <typename T>
+mat Family::bfgs(const T& x, const mat&y, const double rho, const mat& u){
+  uword p = x.n_cols;
+  mat z(u);
+  mat g(u);
+  mat I(p,p,fill::eye);
+
+  mat h(I);
+
+  uword max_iter = 1000;
+  double tolerance = 1e-6;
+
+  uword iter;
+
+  for(iter = 0; iter < max_iter; iter++){
+
+    mat lin_pred = x*z;
+    g = gradient(x,y,lin_pred) + rho*(z-u);
+
+    if(norm(g)<tolerance) break;
```

Comment:
please add a space after `if` statements and separate relational operators with whitespace
```suggestion
    if (norm(g) < tolerance)
      break;
```